src/tz_macos.rs

```diff
@@ -1,80 +1,132 @@
+            }
+        }
 
-impl<T> Drop for Dropping<T> {
-    #[inline]
-    fn drop(&mut self) {
-        unsafe { CFRelease(self.0 as CFTypeRef) };
+        pub(crate) fn name(&self) -> Option<super::string_ref::StringRef<Self>> {
+            // SAFETY: `SystemTimeZone` is only ever created with a valid `CFTimeZoneRef`.
+            let string = unsafe { CFTimeZoneGetName(self.0) };
+            if string.is_null() {
+                None
+            } else {
+                Some(super::string_ref::StringRef::new(string, self))
+            }
+        }
     }
 }
 
-impl<T> Dropping<T> {
-    #[inline]
-    unsafe fn new(v: *const T) -> Option<Self> {
-        if v.is_null() {
+mod string_ref {
+    //! create safe wrapper around `CFStringRef`
+
+    use core_foundation_sys::{
+        base::{Boolean, CFRange},
+        string::{
+            kCFStringEncodingUTF8, CFStringGetBytes, CFStringGetCStringPtr, CFStringGetLength,
+            CFStringRef,
+        },
+    };
+
+    pub(crate) struct StringRef<'a, T> {
+        string: CFStringRef,
+        _parent: &'a T,
+    }
+
+    impl<'a, T> StringRef<'a, T> {
+        pub(crate) fn new(string: CFStringRef, _parent: &'a T) -> Self {
+            Self { string, _parent }
+        }
+
+        pub(crate) fn as_utf8(&self) -> Option<&'a str> {
+            // SAFETY: `StringRef` is only ever created with a valid `CFStringRef`.
+            let v = unsafe { CFStringGetCStringPtr(self.string, kCFStringEncodingUTF8) };
+            if !v.is_null() {
+                // SAFETY: `CFStringGetCStringPtr()` returns NUL-terminated strings.
+                let v = unsafe { std::ffi::CStr::from_ptr(v) };
+                if let Ok(v) = v.to_str() {
+                    return Some(v);
+                }
+            }
             None
-        } else {
-            Some(Dropping(v))
+        }
+
+        pub(crate) fn to_utf8<'b>(&self, buf: &'b mut [u8]) -> Option<&'b str> {
+            // SAFETY: `StringRef` is only ever created with a valid `CFStringRef`.
+            let length = unsafe { CFStringGetLength(self.string) };
+
+            let mut buf_bytes = 0;
+            let range = CFRange {
+                location: 0,
+                length,
+            };
+
+            let converted_bytes = unsafe {
+                // SAFETY: `StringRef` is only ever created with a valid `CFStringRef`.
+                CFStringGetBytes(
+                    self.string,
+                    range,
+                    kCFStringEncodingUTF8,
+                    b'\0',
+                    false as Boolean,
+                    buf.as_mut_ptr(),
+                    buf.len() as isize,
+                    &mut buf_bytes,
+                )
+            };
+            if converted_bytes != length || buf_bytes < 0 || buf_bytes as usize > buf.len() {
+                None
+            } else {
+                std::str::from_utf8(&buf[..buf_bytes as usize]).ok()
+            }
```

Comment:
This breaks up the conditional a bit and removes the duplicate len check and panicking index operation.

```suggestion
            if converted_bytes != length {
                return None;
            }
            let len = usize::try_from(buf_bytes).ok()?;
            let s = buf.get(..len)?;
            std::str::from_utf8(s).ok()
```

Reply:
Probably need a `use std::convert::TryFrom;` to make this work.

Reply:
thanks, yes, this is much nicer. Pushed in 55ddac33aca8aa4e286c05d16d9b3478995cb6c5. Thanks @lopopolo

Reply:
Drat, `std::convert::TryFrom` was introduced in rust 1.34, whereas our MSRV is 1.31, so I had to revert this.

Reply:
Ahh that's unfortunate.

`chrono` 0.4.x's MSRV is currently 1.38. Do you think this would be worth considering an MSRV bump to 1.34?

Reply:
we bumped MSRV in #70 which allowed us to make this change.